pkg/governance/provider.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package governance
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/trustbloc/edge-core/pkg/log"
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+var logger = log.New("edge-adapter/governance")
+
+const (
+	governanceProfileName         = "governance"
+	issueCredentialURLFormat      = "%s/%s" + "/credentials/issueCredential"
+	storeName                     = "governance"
+	governanceVCKey               = "governance_vc"
+	vcsGovernanceRequestTokenName = "vcs_governance" //nolint: gosec
+)
+
+// TODO generate vc from governance vcs https://github.com/trustbloc/edge-service/issues/469
+// nolint: gochecknoglobals
+var governanceJSONLdDoc = `{
+ "@context": [
+   "https://www.w3.org/2018/credentials/v1",
+   "https://trustbloc.github.io/context/governance/context.jsonld"
+ ],
+ "type": "VerifiableCredential",
+ "id": "http://governance.gov/credentials/3732",
+ "issuanceDate": "2020-03-16T22:37:26.544Z",
+ "issuer": {
+   "id": "did:example:oakek12as93mas91220dapop092",
+   "name": "Governance"
+ },
+ "credentialSubject": {
+   "name": "trustbloc",
+   "version": "1.0",
+   "logo": "https://example.com/logo",
+   "description": "Trustbloc Governs.",
+   "docs_uri": "https://example.com/docs",
+   "data_uri": "https://example.com/data.json",
+   "topics": ["banking"],
+   "jurisdictions": ["ca"],
+   "geos": ["Canadian"],
+   "roles": ["accreditor"],
+   "privileges": [
+       {"name": "accredit", "uri": "https://example.com/accredit"}
+   ],
+  "duties": [
+       {"name": "safe-accredit", "uri": "https://example.com/responsible-accredit"}
+    ],
+   "define": [
+       {"name": "DID", "id": "%s"}
```

Comment:
the did ID will likely be placed somewhere else in the future, maybe credentialSubject.id